tests/end-to-end/simple/App.java

```diff
@@ -0,0 +1,185 @@
+package com.acme.test.simple;
+
+
+import software.constructs.Construct;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import software.amazon.awscdk.*;
+import software.amazon.awscdk.App;
+import software.amazon.awscdk.CfnMapping;
+import software.amazon.awscdk.CfnTag;
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+
+
+import software.amazon.awscdk.services.s3.*;
+import software.amazon.awscdk.services.sqs.*;
+
+class GenericList<T> extends LinkedList<T> {
+    public GenericList<T> extend(final T object) {
+        this.addLast(object);
+        return this;
+    }
+
+    public GenericList<T> extend(final List<T> collection) {
+        this.addAll(collection);
+        return this;
+    }
+}
+
+class GenericMap<T, S> extends HashMap<T, S> {
+    public GenericMap<T, S> extend(final T key, final S value) {
+        this.put(key, value);
+        return this;
+    }
+
+    public List<CfnTag> getTags() {
+        final List<CfnTag> tags = new LinkedList<>();
+        for (Map.Entry<T, S> entry : this.entrySet()) {
+            tags.add(CfnTag.builder()
+                    .key(String.valueOf(entry.getKey()))
+                    .value(String.valueOf(entry.getValue()))
+                    .build());
+        }
+        return tags;
+    }
+}
+
+class Mapping<T> {
+    private final String name;
+    private final Construct scope;
+    private final Map<String, Map<String, T>> inner = new TreeMap<>();
+
+    public Mapping(Construct scope, String name) {
+        this.name = name;
+        this.scope = scope;
+    }
+
+    public Mapping<T> put(String primaryKey, String secondaryKey, T value) {
+        final Map<String, T> map = inner.getOrDefault(primaryKey, new TreeMap<>());
+        map.put(secondaryKey, value);
+        inner.put(primaryKey, map);
+        return this;
+    }
+
+    public CfnMapping get() {
+        return CfnMapping.Builder.create(this.scope, this.name).mapping(this.inner).build();
+    }
+}
```

Comment:
This probably should be moved _after_ the actual business the customer is here for (that is, the app & stack).